### PR TITLE
Feat: watchlater

### PR DESCRIPTION
**Pull request by yashpatel1797**
Feature added
1. add video to watch later
2. remove video from watch later

Login is required to add video in watch later.

### CODE REVIEW

File: src/components/PlayListModal/PlayListModal.js [hunk 1]
```diff
@@ -23,6 +23,13 @@ const PlayListModal = ({ video }) => {
...
         }
     }
+    const watchListHandler = () => {
+        if (isVideoInWatchLater(video._id, watchLater)) {
+            removeVideoFromWatchLater(_id, playListDispatch);
+        } else {
+            addVideoInWatchLater(video, playListDispatch);
+        }
+    }
     return ReactDOM.createPortal(
```
Comment (ivinayakg) [inline src/components/PlayListModal/PlayListModal.js:33]: nice that you have used this.

File: src/context/playlist-context.js [hunk 2]
```diff
@@ -1,28 +1,30 @@
 import React, { useState, useContext, createContext, useReducer, useEffect } from "react"
 import { playListReducer, playListTitleReducer } from "reducer";
-import { getPlaylistData } from "utilities"
+import { getPlaylistData, getWatchLaterData } from "utilities"
 
 const PlayListContext = createContext();
 
 const PlayListProvider = ({ children }) => {
 
     const [showModal, setShowModal] = useState(false);
```
Comment (ivinayakg) [inline src/context/playlist-context.js:9]: this can be shifted to the modal itself

File: src/context/playlist-context.js [hunk 3]
```diff
@@ -1,28 +1,30 @@
...
         allPlayListTitle: [],
         playListTitle: []
     })
-    const { playlists } = playListState
+    const { playlists, watchLater } = playListState
     const { playListTitle, allPlayListTitle } = playListTitleState
 
     useEffect(() => {
         getPlaylistData(playListDispatch);
+        getWatchLaterData(playListDispatch);
```
Comment (ivinayakg) [inline src/context/playlist-context.js:23]: no need to fire this, as this is mockbee, although great that your code is capable for future adaptation

**Review by ivinayakg — APPROVED** [review]
great work, though there were times i got a bit confused but nothing to worry, your code is understandable, Great Work

**Review by haarsh24 — APPROVED** [review]
1. UI Looks great
2. Clean file structure
3. Good to merge

**Pull request merged**